### PR TITLE
(proposal) adds a management command for converting all tables to innodb

### PR DESCRIPTION
**Pull request by rosskarchner**
## Additions

- a new management command, convert_to_innodb, which ensures all Django tables are in innodb.
## Testing

1. run `manage.py convert_to_innodb` after importing a recent dump
2. `manage.py dbshell` to enter the MySQL Shell, and `SHOW TABLE STATUS` will allow you to confirm that the deed is done.

## Checklist

* [ ] PR has an informative and human-readable title
* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated
* [ ] Reviewers requested with the [Assignee tool](https://help.github.com/articles/assigning-issues-and-pull-requests-to-other-github-users/) :arrow_right:

### CODE REVIEW

**Comment by higs4281** [comment]
@rosskarchner thanks for the lightning fix!

I run into this error:

django.db.utils.OperationalError: (1347, "'v1.knowledgebase_question_ratings' is not BASE TABLE")

**Comment by rosskarchner** [comment]
hmm-- I'll look at that. I'm working on another PR that restores the MyISAm default, and then this PR will be for executing the switch

File: cfgov/cfgov/settings/local.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
                 'PASSWORD': os.environ.get('MYSQL_PW', ''),
                 'HOST': os.environ.get('MYSQL_HOST', ''),
                 'PORT': os.environ.get('MYSQL_PORT', ''),
-                'OPTIONS': {'init_command': os.environ.get('STORAGE_ENGINE', 'SET storage_engine=MYISAM') },
```
Comment (higs4281) [inline cfgov/cfgov/settings/local.py]: should the defaults be `'SET default_storage_engine=INNODB'` ?

Comment (rosskarchner) [inline cfgov/cfgov/settings/local.py]: Our prod servers are still running MySQL 5.1 at the moment, and 'storage_engine' still seems like the correct syntax there? I'll experiment to see if SET default_storage_engine works on 5.1 though

Comment (rosskarchner) [inline cfgov/cfgov/settings/local.py]: ah, but this is local.py-- so it wouldn't affect production

**Comment by higs4281** [comment]
The tables that fail have TABLE TYPE of VIEW, not BASE TABLE.

unfortunately, SHOW TABLE STATUS doesn't expose TABLE TYPE.

There are only 4 VIEW tables.

```
+------------------------------------------------------+--------+------------+
| TABLE_NAME                                           | ENGINE | TABLE_TYPE |
+------------------------------------------------------+--------+------------+
| knowledgebase_question_ratings                       | NULL   | VIEW       |
| knowledgebase_question_related_questions             | MyISAM | BASE TABLE |
| knowledgebase_question_submissions                   | NULL   | VIEW       |

...

| vw_jobs_foia_request               | NULL   | VIEW       |
| vw_reg_comments                    | NULL   | VIEW       |

```

File: cfgov/core/management/commands/convert_to_innodb.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    """
+
+    def handle(self, database="default", *args, **options):
+
+        cursor = connections[database].cursor()
+
+        cursor.execute("SHOW TABLE STATUS")
+
+        for row in cursor.fetchall():
+            if row[1] != "InnoDB":
```
Comment (higs4281) [inline cfgov/core/management/commands/convert_to_innodb.py]: how about:
`if row[1] == "MyISAM":`

Comment (rosskarchner) [inline cfgov/core/management/commands/convert_to_innodb.py]: OK, I'll implement this

Comment (rosskarchner) [inline cfgov/core/management/commands/convert_to_innodb.py]: Done!

**Comment by rosskarchner** [comment]
Let's put this on the backlog to do next sprint.

**Comment by chosak** [comment]
Closing this as we're moving from MySQL to PostgreSQL.

**Pull request closed without merging**